#160730115 Create a route for getting a user's order history

davidshare opened the pull request:
#### What does this PR do?
Create a route for getting a user's order history

#### Description of Task to be completed?
- add the endpoint for the route to the routes index file
- create a method in the orders controller to handle responses to the endpoint
- create a method to validate the userId
- write test for the endpoint
- update readme file with newly added endpoint


#### How should this be manually tested?
- clone the repo and navigate to the root directory using the command line
- on the command line type npm install to install the node modules
- on the command line, type 'npm run dev-start'
- start postman and use it to test all the endpoints

#### What are the relevant pivotal tracker stories?
[#160730115](https://www.pivotaltracker.com/story/show/160730115)

davidshare committed:
feature(orderHistory): route to get order history

- add the endpoint for the route to the routes index file
- create a method in the orders controller to handle responses to the endpoint
- create a method to validate the userId
- write test for the endpoint
- update readme file with newly added endpoint

Finishes #160730115

davidshare committed:
add authentication to orders route

KaiserPhemi commented on server/middleware/userValidator.js:
```diff
@@ -107,6 +107,27 @@ class ValidateUser {
+   */
+  static validateUserId(request, response, next) {
+    if (!ValidationHelper.checkValidId(request, request.params.userId)) {
+      console.log('here+++++++++++++++++++++++++++++++++++++++++++++');
```
#### Nitpick
There should be no instance of `console.log()` in your code.

davidshare replied:
Thank you! missed that one

davidshare replied:
Changes have been effected

davidshare committed:
bug(tests): fix bug with test for admin functions

- add a query to insert a new admin during database migration
- refactor tests suites to perform admin login
- remove console.log statements
